### PR TITLE
Add Title I endpoints

### PR DESCRIPTION
**Pull request by ywkim312**
After running the main.py test following three endpoints 

- http://localhost:5000/pdl/programs/commodities/map
- http://localhost:5000/pdl/programs/commodities/subprograms
- http://localhost:5000/pdl/programs/commodities/state-distribution

### CODE REVIEW

File: app/controllers/data/commodities/commodities_map_data.json
Comment (pengyin-shan) [inline app/controllers/data/commodities/commodities_map_data.json:1]: In the last meeting, Johnathan mentioned he also wanted a bar chart similar to SNAP. So I wonder is it possible to get a year-based structure that has a similar format to the year-based data in SNAP? Like the following format:
```
{"2018": ["Alabama":[{"programs": [
        {
          "programName": "Agriculture Risk Coverage (ARC)",
          "subPrograms": [
            {
              "subProgramName": "Agriculture Risk Coverage County Option (ARC-CO)",
              "paymentInDollars": 21397138.64
            },
            {
              "subProgramName": "Agriculture Risk Coverage Individual Coverage (ARC-IC)",
              "paymentInDollars": 0
            }
          ],
          "programPaymentInDollars": 21397138.64
        },
        {
          "programName": "Price Loss Coverage (PLC)",
          "subPrograms": [],
          "programPaymentInDollars": 0
        },
        {
          "programName": "Dairy Margin Coverage Program (DMC)",
          "subPrograms": [],
          "programPaymentInDollars": 0
        },
        {
          "programName": "Disaster Assistance",
          "subPrograms": [
            {
              "subProgramName": "Emergency Assistance for Livestock, Honeybees, and Farm-Raised Fish (ELAP)",
              "paymentInDollars": 0
            },
            {
              "subProgramName": "Livestock Forage Disaster Program (LFP)",
              "paymentInDollars": 43435871.65
            },
            {
              "subProgramName": "Livestock Indemnity Program (LIP)",
              "paymentInDollars": 127071.0
            },
            {
              "subProgramName": "Noninsured Crop Disaster Assistance Program (NAP)",
              "paymentInDollars": 0
            },
            {
              "subProgramName": "Tree Assistance Program (TAP)",
              "paymentInDollars": 9952.0
            }
          ],
          "programPaymentInDollars": 43572894.65
        }
      ]},
      "totalPaymentInDollars": 64970033.29
},
{"Alaska":...}
],
"2019":[...]
}
```

Thank you!

Comment (ywkim312) [reply app/controllers/data/commodities/commodities_map_data.json:1]: @pengyin-shan The endpoint will only serve the json as it is. We might want to create a json file like format above, or make reformat method in the api. @sandeep-ps what do you think?

Comment (pengyin-shan) [reply app/controllers/data/commodities/commodities_map_data.json:1]: Based on Johnathan's mentioning, so far he only mentioned SNAP and Title I to have charts, so I think it is a case-by-case situation

**Comment by pengyin-shan** [comment]
Hi @sandeep-ps and @ywkim312 , sorry ask more about data points. Based on the new treemap design that Leigh showed and Johnathan approved in last week's meeting, we will need **Total Payments** v.s **Recipients** v.s. **Base Acres** for each state, grouped by year. The finished result will be like this:

Screenshot 2023-05-11 at 9 35 06 AM

For now, it would be very beneficial if we can have ARC, ARC-CO and ARC-IC data, for 2018 to 2022 with the following structure:
```
{
  "2018-2022":
    { "IL":
     {"Total Payments": 000000,
      "Base Acres(ac)": 00000,
      "Recipients": 000000
      },
      "IN":
     {"Total Payments": 000000,
      "Base Acres(ac)": 00000,
      "Recipients": 000000
      },
}
```
I'm sorry to bother you both, but I don't know if we already have JSON, which needs to transfer to API, or if we still need to transfer Excel to JSON, thus I am asking all of you. Please let me know if there are any questions or if you like me to make this a separate ticket. Thank you!

File: app/controllers/data/commodities/commodities_map_data.json
Comment (pengyin-shan) [reply app/controllers/data/commodities/commodities_map_data.json:1]: Sorry @sandeep-ps , based on the result of the last client meeting, we will change the chart to a treemap, and the new data structure is needed. Please see my comments below.

**Comment by sandeep-ps** [comment]
Hi, @pengyin-shan, I had started with updating the Title 1 JSON structure but later got sidetracked with CSP updates and the release. I will get back to this task soon. Just to clarify, you need https://github.com/policy-design-lab/pdl-api/pull/52/files#diff-7811ce026c1de55d271c37626cfd3d634a701b82fc2da254de8dee3a59aae54f to be updated to include Base Acres (ac) and Recipients for each state, right? I will look at the excel documents for those fields.

**Comment by pengyin-shan** [comment]
> Hi, @pengyin-shan, I had started with updating the Title 1 JSON structure but later got sidetracked with CSP updates and the release. I will get back to this task soon. Just to clarify, you need https://github.com/policy-design-lab/pdl-api/pull/52/files#diff-7811ce026c1de55d271c37626cfd3d634a701b82fc2da254de8dee3a59aae54f to be updated to include Base Acres (ac) and Recipients for each state, right? I will look at the excel documents for those fields.

Yes, I will need Base Acres (ac) and Recipients for each state. Currently, we need 2018-2022 for ARC, ARC-CO, and ARC-I IC programs.

In the future, we have a possibility to add this data structure to other Commiidty I programs such as PLC, but I'll confirm with the designer first and then let you know. Thank you!

**Comment by sandeep-ps** [comment]
Hi, @ywkim312, Next week, you could use the JSON files from this folder for to update this PR. Thanks.

**Comment by ywkim312** [comment]
> Hi, @ywkim312, Next week, you could use the JSON files from [this](https://github.com/policy-design-lab/data-import/tree/develop/title-1-commodities) folder for to update this PR. Thanks.

got it, thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,8 +40,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 - EQIP json GET endpoints. [#35](https://github.com/policy-design-lab/pdl-api/issues/35)
-- Title II CSP json GET endpoints. [#45](https://github.com/policy-design-lab/pdl-api/issues/45)
+- Title I Commodities json GET endpoints. [#40](https://github.com/policy-design-lab/pdl-api/issues/40)
```
Comment (sandeep-ps) [inline CHANGELOG.md:43]: It looks like a merge may have updated a previous entry. Please fix the CHANGELOG entry and move this to the unreleased section.

**Review by sandeep-ps — APPROVED** [review]
I tested the endpoints using Postman. It looks good to me. The CHANGELOG change needs to be fixed (left a comment for that). Approving.

File: CHANGELOG.md (hunk 1 above)
Comment (ywkim312) [reply CHANGELOG.md]: fixed it

**Review by pengyin-shan — APPROVED** [review]
Once docker test is passed, pls feel free to merge this PR.

**Pull request merged**